lib/chef/resource/windows_certificate.rb

```diff
@@ -58,7 +58,21 @@ class WindowsCertificate < Chef::Resource
 
       action :create do
         description "Creates or updates a certificate."
-        add_cert(OpenSSL::X509::Certificate.new(raw_source))
+
+        cert_obj = OpenSSL::X509::Certificate.new(raw_source) # A certificate object in memory
+        thumbprint = OpenSSL::Digest::SHA1.new(cert_obj.to_der).to_s # Fetch its thumbprint
+
+        # Check whether a certificate with this thumbprint
+        # is already present in certificate store
+        exists = verify_cert(thumbprint)
+
+        if (!!exists == exists) && exists
```

Comment:
whatever this line of code is doing it absolutely needs to be made clearer

Reply:
Yes, actually I also didn't like the way `verify_cert` returns the output. When a certificate for the given thumbprint is not present, then it returns a `String`: "Certificate not found" from [this line of code](https://github.com/chef/win32-certstore/blob/master/lib/win32/certstore/store_base.rb#L174). But when a certificate is present, it returns `Boolean`: true.
Due to this reason, I applied the same [logic which is used in `:verify` action](https://github.com/chef/chef/blob/master/lib/chef/resource/windows_certificate.rb#L114-L117) with a thought that may be it was an acceptable solution.
As per the logic, if the return output is of Boolean type (!!out == out) AND its value is true, then certificate is present in the store.
I can put the comments to make it clear, but will it not be better to changes such tricky unplanned returns in library itself. Need your views. Thanks.